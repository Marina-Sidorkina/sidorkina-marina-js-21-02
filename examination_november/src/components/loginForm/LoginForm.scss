.login {
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 500px;
  padding: 15px;
  margin-top: 50px;
  background-color: #ffffff;
  border-radius: 15px;

  &__title {
    text-align: center;
  }

  &__form {
    width: 500px;
    height: 190px;

    @media (max-width: 555px) {
      width: 100%;
    }
  }

  &__form-item,
  &__button,
  &__submit,
  &__input {
    width: 500px;
    min-width: 260px;

    @media (max-width: 555px) {
      width: 100%;
    }
  }

  &__button {
    background-color: #000000;
    color: #ffffff;
    border: none;
    border-radius: 5px;
  }

  &__input {
    border-radius: 5px;
  }

  &__link {
    text-align: center;
    color: #808080;
  }
}