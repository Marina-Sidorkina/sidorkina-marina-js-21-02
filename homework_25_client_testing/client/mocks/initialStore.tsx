const initialStore = {
  userModal: {
    isOpened: false,
    values: {
      picture: '',
      name: 'John Doe',
      gender: 'male',
      dateOfBirth: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      registerDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      id: '12345',
      email: 'test',
      phone: '55555'
    }
  },
  login: {
    data: {
      error: false,
      isLoading: false,
      inputValue: '',
      authorizedUserId: '123345',
      authorizedUserName: 'Name',
      authorizedUserPicture: ''
    }
  },
  registration: {
    values: {
      name: '',
      gender: '',
      dateOfBirth: '',
      email: '',
      phone: '',
    },
    isLoading: false,
    error: false
  },
  languageSelector: {
    value: 'ru'
  },
  postModal: {
    isOpened: false,
    currentPostId: '12345'
  },
  userPosts: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  postsList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  usersList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      users: [{
        firstName: 'John',
        lastName: 'Doe',
        title: 'mr',
        id: '12345'
      }]
    }
  },
  userInfo: {
    data: {
      isLoading: false,
      user: {
        firstName: 'John',
        lastName: 'Doe',
        picture: '',
        title: 'mr',
        gender: 'male',
        dateOfBirth: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        registerDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        id: '12345',
        email: '',
        phone: '55555'
      }
    }
  },
  postModalPost: {
    isLoading: false,
    error: false,
    owner: {
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    },
    post: {
      id: '12345',
      image: '',
      text: 'text',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }
  },
  postModalComments: {
    page: 1,
    error: false,
    isLoading: false,
    comments: [{
      id: '1234',
      message: 'message',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }],
    owners: [{
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    }],
    limit: 6,
    totalComments: 0
  }
};

export default initialStore;
