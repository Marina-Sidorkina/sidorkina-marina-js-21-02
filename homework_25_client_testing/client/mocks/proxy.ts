import MockAdapter from 'axios-mock-adapter';
import { proxy } from '../src/api/proxy';

export const mock = new MockAdapter(proxy);
