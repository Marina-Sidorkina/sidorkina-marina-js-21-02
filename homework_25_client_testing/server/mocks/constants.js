module.exports = {
  errorText: 'Request failed with status code 520',
  successStatus: 200,
  errorStatus: 520
};
