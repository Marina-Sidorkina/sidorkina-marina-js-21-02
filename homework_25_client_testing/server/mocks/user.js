module.exports = {
  userListMockData: { data: [], page: 0, limit: 5, total: 100 },
  userMockData: {
    id: '60d0fe4f5311236168a109ce',
    title: 'mr',
    firstName: 'Rudi',
    lastName: 'Droste',
    picture: 'https://randomuser.me/api/portraits/med/men/83.jpg',
    gender: 'male',
    email: 'rudi.droste@example.com',
    dateOfBirth: '2021-06-21T21:02:08.029Z',
    phone: '0421-7342787',
    location: {
      street: '1196, MÃ¼hlenweg',
      city: 'GrÃ¼nhain-Beierfeld',
      state: 'Sachsen-Anhalt',
      country: 'Germany',
      timezone: '+3:30'
    },
    registerDate: '2021-06-21T21:02:08.029Z',
    updatedDate: '2021-06-21T21:02:08.029Z'
  },
  userProcessedMockData: {
    id: '60d0fe4f5311236168a109ce',
    firstName: 'Rudi',
    lastName: 'Droste',
    title: 'mr',
    gender: 'male',
    email: 'rudi.droste@example.com',
    phone: '0421-7342787',
    dateOfBirth: {
      enDate: '22 of June 2021',
      ruDate: '22 Июня 2021 года',
      enDateAndTime: '22 of June 00:02',
      ruDateAndTime: '22 Июня 00:02'
    },
    registerDate: {
      enDate: '22 of June 2021',
      ruDate: '22 Июня 2021 года',
      enDateAndTime: '22 of June 00:02',
      ruDateAndTime: '22 Июня 00:02'
    },
    picture: 'https://randomuser.me/api/portraits/med/men/83.jpg'
  },
  userPostsListMockData: {
    data: [{id: '12345', image: 'image', owner: {}, text: 'text'}],
    page: 0, limit: 1, total: 100
  },
  userUpdatedMockData: { firstName: 'Rudi', lastname: 'Droste', gender: 'male',
    email: 'rudi.droste@example.com', dateOfBirth: '2021-06-21T21:02:08.029Z',
    phone: '0421-7342787'
  }
};
