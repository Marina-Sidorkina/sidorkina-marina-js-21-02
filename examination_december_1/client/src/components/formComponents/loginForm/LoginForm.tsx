import React, { useContext, useEffect } from 'react';
import '../../../locale/i18next';
import {
  Form, Input, Button, Spin
} from 'antd';
import { useHistory } from 'react-router';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import {
  authorizeUser, hideLoadingAction,
  resetAuthorizationErrorAction, updateAuthorizationInputValue
} from '../../../redux/actions/login';
import styles from './LoginForm.module.scss';
import { ThemeContext } from '../../../contexts/ThemeContext';
import { RULES, RULES_EN } from './antDesignSettings/loginForm';
import { LOGIN_INPUT_FILED_NAME, RUSSIAN_LANGUAGE } from '../../../constants/components';
import { useTypedSelector } from '../../../redux/hooks/useTypedSelector';

const LoginForm = () => {
  const history = useHistory();
  const themeContext = useContext(ThemeContext);
  const { t } = useTranslation();
  const stateValues = useTypedSelector((state) => state);
  const dispatch = useDispatch();
  const language = useTypedSelector((state) => state.languageSelector.value);
  const rules = language === RUSSIAN_LANGUAGE ? RULES : RULES_EN;

  const onSubmit = (values: any) => {
    dispatch(authorizeUser(values[LOGIN_INPUT_FILED_NAME], history));
  };

  const onChange = (evt: any) => {
    dispatch(updateAuthorizationInputValue(evt.target.value));
  };

  useEffect(() => {
    dispatch(resetAuthorizationErrorAction());
    return () => {
      dispatch(hideLoadingAction());
    };
  }, []);

  return (
    <div className={themeContext.darkTheme ? `${styles.login} ${styles.login_dark}` : styles.login}>
      <Form
        className={styles.form}
        onFinish={onSubmit}
      >
        <h2 className={styles.title}>
          { t('login.title', {}) }
        </h2>
        <Form.Item className={`${styles.formItem} ${styles.formItem_add}`}>
          <label className={styles.label}>ID:</label>
          <Form.Item
            className={styles.formItem}
            name="login-input"
            rules={rules.id}
          >
            <Input
              className={styles.input}
              value={stateValues.login.data.inputValue}
              placeholder={t('login.placeholder', {})}
              onChange={onChange}
            />
          </Form.Item>
        </Form.Item>

        <Form.Item className={styles.submit}>
          <Button
            className={styles.button}
            type="primary"
            htmlType="submit"
          >
            { t('login.button', {}) }
          </Button>
          {stateValues.login.data.isLoading
            ? (
              <Spin
                size="small"
                style={{
                  width: '110px',
                  height: '110px',
                  position: 'absolute',
                  top: '-21px',
                  left: 'calc(50% - 55px)'
                }}
              />
            ) : null}
          {stateValues.login.data.error
            ? (
              <div className={styles.error}>
                { t('login.error', {}) }
              </div>
            )
            : null}
        </Form.Item>
      </Form>
      <Link to="registration">
        <span className={styles.link}>
          { t('login.link', {}) }
        </span>
      </Link>
    </div>
  );
};

export default LoginForm;
