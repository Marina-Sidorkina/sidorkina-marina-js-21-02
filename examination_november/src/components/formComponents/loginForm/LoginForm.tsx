import React, { useContext, useEffect } from 'react';
import './LoginForm.scss';
import {
  Form, Input, Button, Spin
} from 'antd';
import { useHistory } from 'react-router';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import {
  authorizeUser,
  hideLoadingAction,
  resetAuthorizationErrorAction,
  updateAuthorizationInputValue
} from '../../../redux/actions/login';
import { ThemeContext } from '../../../contexts/ThemeContext';
import { RULES } from '../../../antDesignSettings/loginForm';
import { LOGIN_INPUT_FILED_NAME } from '../../../constants/components';
import { ILoginFormProps } from './@types/loginForm';

const LoginForm = (props: ILoginFormProps) => {
  const history = useHistory();

  const {
    isLoading, authorize, updateInputValue, inputValue,
    hideLoading, error, resetError
  } = props;

  const themeContext = useContext(ThemeContext);

  const onSubmit = (values: any) => {
    authorize(values[LOGIN_INPUT_FILED_NAME], history);
  };

  const onChange = (evt: any) => {
    updateInputValue(evt.target.value);
  };

  useEffect(() => {
    resetError();
    return () => hideLoading();
  }, []);

  return (
    <div className={`${themeContext.darkTheme
      ? 'login login_dark'
      : 'login'}`}
    >
      <Form
        className="login__form"
        onFinish={onSubmit}
      >
        <h2 className="login__title">Вход</h2>
        <Form.Item className="login__form-item login__form-item_add">
          <label className="login__label">ID:</label>
          <Form.Item
            className="login__form-item"
            name="login-input"
            rules={RULES.id}
          >
            <Input
              className="login__input"
              value={inputValue}
              placeholder="Введите свой ID"
              onChange={onChange}
            />
          </Form.Item>
        </Form.Item>

        <Form.Item className="login__submit">
          <Button
            className="login__button"
            type="primary"
            htmlType="submit"
          >
            Войти
          </Button>
          {isLoading
            ? (
              <Spin
                className="login__spinner"
                size="small"
                style={{
                  width: '110px',
                  height: '110px',
                  position: 'absolute',
                  top: '-21px',
                  left: 'calc(50% - 55px)'
                }}
              />
            ) : null}
          {error ? <div className="login__error">ID не существует</div> : null}
        </Form.Item>
      </Form>
      <Link to="registration">
        <span className="login__link">Ещё нет аккаунта? Зарегистрироваться</span>
      </Link>
    </div>
  );
};

export default connect(
  (state: any) => ({
    isLoading: state.login.data.isLoading,
    inputValue: state.login.data.inputValue,
    error: state.login.data.error,
  }),
  (dispatch) => ({
    authorize: bindActionCreators(authorizeUser, dispatch),
    updateInputValue: bindActionCreators(updateAuthorizationInputValue, dispatch),
    hideLoading: bindActionCreators(hideLoadingAction, dispatch),
    resetError: bindActionCreators(resetAuthorizationErrorAction, dispatch)
  })
)(LoginForm);
