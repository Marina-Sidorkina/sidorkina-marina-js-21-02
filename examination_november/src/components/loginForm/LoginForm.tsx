import React, { useEffect } from 'react';
import './LoginForm.scss';
import {
  Form, Input, Button, Spin
} from 'antd';
import { useHistory } from 'react-router';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  authorizeUser,
  hideLoadingAction,
  resetAuthorizationErrorAction,
  updateAuthorizationInputValue
} from '../../redux/actions/login';

interface ILoginFormProps {
  isLoading: boolean;
  authorize: Function;
  updateInputValue: Function;
  inputValue: string;
  hideLoading: Function;
  error: any;
  resetError: Function;
}

const LoginForm = (props: ILoginFormProps) => {
  const history = useHistory();
  const {
    isLoading, authorize, updateInputValue, inputValue,
    hideLoading, error, resetError
  } = props;

  const onSubmit = (values: any) => {
    authorize(values['login-input'], history);
  };

  const onChange = (evt: any) => {
    updateInputValue(evt.target.value);
  };

  useEffect(() => {
    resetError();
    return () => hideLoading();
  }, []);

  return (
    <div className="login">
      <Form
        className="login__form"
        onFinish={onSubmit}
      >
        <h2 className="login__title">Вход</h2>
        <Form.Item className="login__form-item login__form-item_add">
          <label className="login__label">ID:</label>
          <Form.Item
            className="login__form-item"
            name="login-input"
            rules={[
              { required: true, message: 'Please enter your ID' }
            ]}
          >
            <Input
              className="login__input"
              value={inputValue}
              placeholder="Введите свой ID"
              onChange={onChange}
            />
          </Form.Item>
        </Form.Item>

        <Form.Item className="login__submit">
          <Button
            className="login__button"
            type="primary"
            htmlType="submit"
          >
            Войти
          </Button>
          {isLoading
            ? (
              <Spin
                className="users-info__spinner"
                size="small"
                style={{
                  width: '110px',
                  height: '110px',
                  position: 'absolute',
                  top: '-21px',
                  left: 'calc(50% - 55px)'
                }}
              />
            ) : null}
          {error ? <div className="login__error">ID не существует</div> : null}
        </Form.Item>
      </Form>
      <a className="login__link" href="login">Ещё нет аккаунта? Зарегистрироваться</a>
    </div>
  );
};

export default connect(
  (state: any) => ({
    isLoading: state.login.data.isLoading,
    inputValue: state.login.data.inputValue,
    error: state.login.data.error,
  }),
  (dispatch) => ({
    authorize: bindActionCreators(authorizeUser, dispatch),
    updateInputValue: bindActionCreators(updateAuthorizationInputValue, dispatch),
    hideLoading: bindActionCreators(hideLoadingAction, dispatch),
    resetError: bindActionCreators(resetAuthorizationErrorAction, dispatch)
  })
)(LoginForm);
