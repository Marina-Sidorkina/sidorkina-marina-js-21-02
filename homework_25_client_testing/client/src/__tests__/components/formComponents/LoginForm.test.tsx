import '../../../../mocks/matchMediaMock';
import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import mockStore from '../../../../mocks/mockStore';
import LoginForm from '../../../components/formComponents/loginForm/LoginForm';
import initialStore from '../../../../mocks/initialStore';

jest.mock('../../../redux/actions/login');
configure({ adapter: new Adapter() });

describe('LoginForm component test', () => {
  let store: any;
  let wrapper: any;

  beforeEach(() => {
    store = mockStore(initialStore);
    store.dispatch = jest.fn();
    wrapper = mount(<Provider store={store}><HashRouter><LoginForm /></HashRouter></Provider>);
  });

  test('should render LoginForm', () => {
    expect(wrapper.length).toEqual(1);
  });

  test('should render LoginForm', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
