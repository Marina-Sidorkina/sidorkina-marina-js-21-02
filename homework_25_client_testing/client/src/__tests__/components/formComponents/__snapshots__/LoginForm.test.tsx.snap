// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component test should render LoginForm 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <LoginForm>
        <div
          className="login"
        >
          <ForwardRef(InternalForm)
            className="form"
            onFinish={[Function]}
          >
            <SizeContextProvider>
              <ForwardRef(Form)
                className="ant-form ant-form-horizontal form"
                form={
                  Object {
                    "__INTERNAL__": Object {
                      "itemRef": [Function],
                      "name": undefined,
                    },
                    "getFieldError": [Function],
                    "getFieldInstance": [Function],
                    "getFieldValue": [Function],
                    "getFieldWarning": [Function],
                    "getFieldsError": [Function],
                    "getFieldsValue": [Function],
                    "getInternalHooks": [Function],
                    "isFieldTouched": [Function],
                    "isFieldValidating": [Function],
                    "isFieldsTouched": [Function],
                    "isFieldsValidating": [Function],
                    "resetFields": [Function],
                    "scrollToField": [Function],
                    "setFields": [Function],
                    "setFieldsValue": [Function],
                    "submit": [Function],
                    "validateFields": [Function],
                  }
                }
                onFinish={[Function]}
                onFinishFailed={[Function]}
              >
                <form
                  className="ant-form ant-form-horizontal form"
                  onReset={[Function]}
                  onSubmit={[Function]}
                >
                  <h2
                    className="title"
                  >
                    Sign in
                  </h2>
                  <FormItem
                    className="formItem formItem_add"
                  >
                    <Row
                      className="ant-form-item formItem formItem_add"
                      key="row"
                    >
                      <div
                        className="ant-row ant-form-item formItem formItem_add"
                        style={Object {}}
                      >
                        <FormItemLabel
                          className="formItem formItem_add"
                          prefixCls="ant-form"
                          requiredMark={true}
                        />
                        <FormItemInput
                          className="formItem formItem_add"
                          errors={Array []}
                          name={Array []}
                          prefixCls="ant-form"
                          status=""
                          touched={false}
                          validateStatus=""
                          validating={false}
                          warnings={Array []}
                        >
                          <Col
                            className="ant-form-item-control"
                          >
                            <div
                              className="ant-col ant-form-item-control"
                              style={Object {}}
                            >
                              <div
                                className="ant-form-item-control-input"
                              >
                                <div
                                  className="ant-form-item-control-input-content"
                                >
                                  <label
                                    className="label"
                                  >
                                    ID:
                                  </label>
                                  <FormItem
                                    className="formItem"
                                    name="login-input"
                                    rules={
                                      Array [
                                        Object {
                                          "message": "Укажите ваш ID",
                                          "required": true,
                                        },
                                      ]
                                    }
                                  >
                                    <WrapperField
                                      className="formItem"
                                      messageVariables={
                                        Object {
                                          "label": "login-input",
                                        }
                                      }
                                      name="login-input"
                                      onMetaChange={[Function]}
                                      rules={
                                        Array [
                                          Object {
                                            "message": "Укажите ваш ID",
                                            "required": true,
                                          },
                                        ]
                                      }
                                      trigger="onChange"
                                      validateTrigger="onChange"
                                    >
                                      <Field
                                        className="formItem"
                                        fieldContext={
                                          Object {
                                            "__INTERNAL__": Object {
                                              "itemRef": [Function],
                                              "name": undefined,
                                            },
                                            "getFieldError": [Function],
                                            "getFieldInstance": [Function],
                                            "getFieldValue": [Function],
                                            "getFieldWarning": [Function],
                                            "getFieldsError": [Function],
                                            "getFieldsValue": [Function],
                                            "getInternalHooks": [Function],
                                            "isFieldTouched": [Function],
                                            "isFieldValidating": [Function],
                                            "isFieldsTouched": [Function],
                                            "isFieldsValidating": [Function],
                                            "resetFields": [Function],
                                            "scrollToField": [Function],
                                            "setFields": [Function],
                                            "setFieldsValue": [Function],
                                            "submit": [Function],
                                            "validateFields": [Function],
                                            "validateTrigger": "onChange",
                                          }
                                        }
                                        key="_login-input"
                                        messageVariables={
                                          Object {
                                            "label": "login-input",
                                          }
                                        }
                                        name={
                                          Array [
                                            "login-input",
                                          ]
                                        }
                                        onMetaChange={[Function]}
                                        rules={
                                          Array [
                                            Object {
                                              "message": "Укажите ваш ID",
                                              "required": true,
                                            },
                                          ]
                                        }
                                        trigger="onChange"
                                        validateTrigger="onChange"
                                        valuePropName="value"
                                      >
                                        <Row
                                          className="ant-form-item formItem"
                                          key="row"
                                        >
                                          <div
                                            className="ant-row ant-form-item formItem"
                                            style={Object {}}
                                          >
                                            <FormItemLabel
                                              className="formItem"
                                              htmlFor="login-input"
                                              name="login-input"
                                              prefixCls="ant-form"
                                              required={true}
                                              requiredMark={true}
                                              rules={
                                                Array [
                                                  Object {
                                                    "message": "Укажите ваш ID",
                                                    "required": true,
                                                  },
                                                ]
                                              }
                                            />
                                            <FormItemInput
                                              className="formItem"
                                              errors={Array []}
                                              name={Array []}
                                              prefixCls="ant-form"
                                              rules={
                                                Array [
                                                  Object {
                                                    "message": "Укажите ваш ID",
                                                    "required": true,
                                                  },
                                                ]
                                              }
                                              status=""
                                              touched={false}
                                              validateStatus=""
                                              validating={false}
                                              warnings={Array []}
                                            >
                                              <Col
                                                className="ant-form-item-control"
                                              >
                                                <div
                                                  className="ant-col ant-form-item-control"
                                                  style={Object {}}
                                                >
                                                  <div
                                                    className="ant-form-item-control-input"
                                                  >
                                                    <div
                                                      className="ant-form-item-control-input-content"
                                                    >
                                                      <Component
                                                        update={
                                                          <Input
                                                            className="input"
                                                            onChange={[Function]}
                                                            placeholder="Enter your ID"
                                                            type="text"
                                                            value=""
                                                          />
                                                        }
                                                      >
                                                        <Input
                                                          className="input"
                                                          id="login-input"
                                                          onChange={[Function]}
                                                          placeholder="Enter your ID"
                                                          type="text"
                                                        >
                                                          <ClearableLabeledInput
                                                            bordered={true}
                                                            className="input"
                                                            element={
                                                              <input
                                                                className="ant-input input"
                                                                id="login-input"
                                                                onBlur={[Function]}
                                                                onChange={[Function]}
                                                                onFocus={[Function]}
                                                                onKeyDown={[Function]}
                                                                placeholder="Enter your ID"
                                                                type="text"
                                                              />
                                                            }
                                                            focused={false}
                                                            handleReset={[Function]}
                                                            id="login-input"
                                                            inputType="input"
                                                            onChange={[Function]}
                                                            placeholder="Enter your ID"
                                                            prefixCls="ant-input"
                                                            triggerFocus={[Function]}
                                                            type="text"
                                                            value=""
                                                          >
                                                            <input
                                                              className="ant-input input"
                                                              id="login-input"
                                                              onBlur={[Function]}
                                                              onChange={[Function]}
                                                              onFocus={[Function]}
                                                              onKeyDown={[Function]}
                                                              placeholder="Enter your ID"
                                                              type="text"
                                                              value=""
                                                            />
                                                          </ClearableLabeledInput>
                                                        </Input>
                                                      </Component>
                                                    </div>
                                                  </div>
                                                  <ErrorList
                                                    className="ant-form-item-explain-connected"
                                                    errors={Array []}
                                                    helpStatus=""
                                                    warnings={Array []}
                                                  >
                                                    <CSSMotion
                                                      motionAppear={false}
                                                      motionDeadline={500}
                                                      motionEnter={false}
                                                      motionName="ant-show-help"
                                                      onAppearActive={[Function]}
                                                      onAppearEnd={[Function]}
                                                      onAppearStart={[Function]}
                                                      onEnterActive={[Function]}
                                                      onEnterEnd={[Function]}
                                                      onEnterStart={[Function]}
                                                      onLeaveActive={[Function]}
                                                      onLeaveEnd={[Function]}
                                                      onLeaveStart={[Function]}
                                                      visible={false}
                                                    >
                                                      <DomWrapper />
                                                    </CSSMotion>
                                                  </ErrorList>
                                                </div>
                                              </Col>
                                            </FormItemInput>
                                          </div>
                                        </Row>
                                      </Field>
                                    </WrapperField>
                                  </FormItem>
                                </div>
                              </div>
                              <ErrorList
                                className="ant-form-item-explain-connected"
                                errors={Array []}
                                helpStatus=""
                                warnings={Array []}
                              >
                                <CSSMotion
                                  motionAppear={false}
                                  motionDeadline={500}
                                  motionEnter={false}
                                  motionName="ant-show-help"
                                  onAppearActive={[Function]}
                                  onAppearEnd={[Function]}
                                  onAppearStart={[Function]}
                                  onEnterActive={[Function]}
                                  onEnterEnd={[Function]}
                                  onEnterStart={[Function]}
                                  onLeaveActive={[Function]}
                                  onLeaveEnd={[Function]}
                                  onLeaveStart={[Function]}
                                  visible={false}
                                >
                                  <DomWrapper />
                                </CSSMotion>
                              </ErrorList>
                            </div>
                          </Col>
                        </FormItemInput>
                      </div>
                    </Row>
                  </FormItem>
                  <FormItem
                    className="submit"
                  >
                    <Row
                      className="ant-form-item submit"
                      key="row"
                    >
                      <div
                        className="ant-row ant-form-item submit"
                        style={Object {}}
                      >
                        <FormItemLabel
                          className="submit"
                          prefixCls="ant-form"
                          requiredMark={true}
                        />
                        <FormItemInput
                          className="submit"
                          errors={Array []}
                          name={Array []}
                          prefixCls="ant-form"
                          status=""
                          touched={false}
                          validateStatus=""
                          validating={false}
                          warnings={Array []}
                        >
                          <Col
                            className="ant-form-item-control"
                          >
                            <div
                              className="ant-col ant-form-item-control"
                              style={Object {}}
                            >
                              <div
                                className="ant-form-item-control-input"
                              >
                                <div
                                  className="ant-form-item-control-input-content"
                                >
                                  <Button
                                    className="button"
                                    htmlType="submit"
                                    type="primary"
                                  >
                                    <Wave
                                      disabled={false}
                                    >
                                      <button
                                        className="ant-btn ant-btn-primary button"
                                        onClick={[Function]}
                                        type="submit"
                                      >
                                        <LoadingIcon
                                          existIcon={false}
                                          loading={false}
                                          prefixCls="ant-btn"
                                        >
                                          <CSSMotion
                                            motionName="ant-btn-loading-icon-motion"
                                            onAppearActive={[Function]}
                                            onAppearStart={[Function]}
                                            onEnterActive={[Function]}
                                            onEnterStart={[Function]}
                                            onLeaveActive={[Function]}
                                            onLeaveStart={[Function]}
                                            removeOnLeave={true}
                                            visible={false}
                                          >
                                            <DomWrapper />
                                          </CSSMotion>
                                        </LoadingIcon>
                                        <span
                                          key=".0"
                                        >
                                          Sign in
                                        </span>
                                      </button>
                                    </Wave>
                                  </Button>
                                </div>
                              </div>
                              <ErrorList
                                className="ant-form-item-explain-connected"
                                errors={Array []}
                                helpStatus=""
                                warnings={Array []}
                              >
                                <CSSMotion
                                  motionAppear={false}
                                  motionDeadline={500}
                                  motionEnter={false}
                                  motionName="ant-show-help"
                                  onAppearActive={[Function]}
                                  onAppearEnd={[Function]}
                                  onAppearStart={[Function]}
                                  onEnterActive={[Function]}
                                  onEnterEnd={[Function]}
                                  onEnterStart={[Function]}
                                  onLeaveActive={[Function]}
                                  onLeaveEnd={[Function]}
                                  onLeaveStart={[Function]}
                                  visible={false}
                                >
                                  <DomWrapper />
                                </CSSMotion>
                              </ErrorList>
                            </div>
                          </Col>
                        </FormItemInput>
                      </div>
                    </Row>
                  </FormItem>
                </form>
              </ForwardRef(Form)>
            </SizeContextProvider>
          </ForwardRef(InternalForm)>
          <Link
            to="registration"
          >
            <LinkAnchor
              href="#/registration"
              navigate={[Function]}
            >
              <a
                href="#/registration"
                onClick={[Function]}
              >
                <span
                  className="link"
                >
                  Don't have an account yet? Sign up now
                </span>
              </a>
            </LinkAnchor>
          </Link>
        </div>
      </LoginForm>
    </Router>
  </HashRouter>
</Provider>
`;
